modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.ts

```diff
@@ -0,0 +1,220 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {TRenderer} from './FrontendDataSetContext';
+
+export declare function FrontendDataSet({
+	actionParameterName,
+	activeViewSettings,
+	apiURL,
+	appURL,
+	bulkActions,
+	creationMenu,
+	currentURL,
+	customDataRenderers,
+	customViews,
+	customViewsEnabled,
+	emptyState,
+	filters,
+	formId,
+	formName,
+	header,
+	id,
+	initialSelectedItemsValues,
+	inlineAddingSettings,
+	inlineEditingSettings,
+	items,
+	itemsActions,
+	namespace,
+	nestedItemsKey,
+	nestedItemsReferenceKey,
+	onActionDropdownItemClick,
+	onBulkActionItemClick,
+	onSelect,
+	overrideEmptyResultView,
+	pagination,
+	portletId,
+	selectedItemsKey,
+	selectionType,
+	showManagementBar,
+	showPagination,
+	showSearch,
+	sidePanelId,
+	sorts,
+	style,
+	views,
+}: IFrontendDataSetProps): JSX.Element;
+
+export declare function DateTimeRenderer({
+	options,
+	value,
+}: DateTimeRendererProps): string;
+
+type DateTimeRendererProps = {
+	options?: {
+		format: {
+			day?: string;
+			hour?: string;
+			minute?: string;
+			month?: string;
+			second?: string;
+			timeZone?: string;
+			year?: string;
+		};
+	};
+	value: string;
+};
+
+type TDelta = {
+	href?: string;
+	label: number;
+};
+
+export interface IInlineEditingSettings {
+	alwaysOn: boolean;
+	defaultBodyContent: object;
+}
+
+export interface IActionsDropdown extends IBaseActions {
+	loading: boolean;
+	menuActive: boolean;
+	onClick: Function;
+	onMenuActiveChange: Function;
+	setLoading: Function;
+}
+
+export interface IBaseActions {
+	actions: IItemsActions[];
+	itemData: any;
+	itemId: number | string;
+}
+
+export interface IItemsActions {
+	data?: IItemActionsData;
+	href?: string;
+	icon?: string;
+	id?: string | number;
+	items?: IItemsActions[];
```

Comment:
On the fly generated property `items` https://github.com/liferay-frontend/liferay-portal/pull/3983/files#diff-bed70a46e4545a7d5c232541bea08157b92a65b58ed561e8b7f3bb229782b3a6R188